xarray/core/accessors.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from .common import is_datetime_like
+from .extensions import register_dataarray_accessor
+from .pycompat import dask_array_type
+
+import numpy as np
+import pandas as pd
+
+
+def _get_date_field(values, name):
+    """Indirectly access pandas' libts.get_date_field by wrapping data
+    as a Series and calling through `.dt` attribute.
+
+    Parameters
+    ----------
+    values : np.ndarray or dask.array-like
+        Array-like container of datetime-like values
+    name : str
+        Name of datetime field to access
+
+    """
+    def _access_through_series(values, name):
```

Comment:
For serialization in dask, it's better to pass top-level functions to `map_blocks`. So I'd just move this to the top level.